code/ship/ship.cpp

```diff
@@ -5610,15 +5610,23 @@ DCF_BOOL( show_velocity_dot, ship_show_velocity_dot )
 
 static bool ballistic_possible_for_this_ship(const ship_info *sip)
 {
-	if (sip->allowed_bank_restricted_weapons.empty())
+	// has no weapons!
+	if (sip->allowed_bank_restricted_weapons.empty() && sip->num_primary_banks < 1)
```

Comment:
Since `sip->allowed_bank_restricted_weapons.empty()` is now being tested below, I would venture to say you can take this out.  This will also be more robust if for whatever reason there were allowed weapons on a ship with no primaries.